sympy/matrices/common.py

```diff
@@ -2065,6 +2065,9 @@ def __pow__(self, num):
                     except (AttributeError, MatrixError):
                         pass
                 return a._eval_pow_by_recursion(num)
+            elif num.is_Number != True and num.is_negative == None and a.det() == 0:
```

Comment:
What should the answer be if num is `var('y',negative=True)`? And did you mean for the `and` to be `or`?

Reply:
If we use ```var( 'y' , negative = True)``` it will result into a value error because negative powers of matrix with det ==0 is not invertible and thats what it is happening here.

It is definitely ``` and ``` because we want if a it is a number and if we dont know whether it is negative or positive then it will proceed into the elif condition.

Reply:
What do you think?